### ISSUE TITLE
Transfer Falling Star to Labnet

### ISSUE THREAD

**Issue by marissaeinhorn** [dev4vater/vater#227]
- [x] Receive packer templates from AFRL
- [x] Verify packer templates run properly - build on morpheus datastore
- [x] Convert terraform file into module
- [x] Test ansible playbooks
- [x] Set up task in semaphore to build exercise & run playbooks
- [x] Test for 2 student teams
- [x] Transfer to Labnet

**Comment by marissaeinhorn** [comment]
` Error: error adding host vSwitch: ServerFaultCode: A specified parameter was not correct:` I took down the vsphere host and its returning this error when i attempt to create new networks

**Comment by marissaeinhorn** [comment]
Virtual switch name was too long or there were random invalid characters.  I cant find any documentation to determine what vsphere virtual switch name requirements are.

**Comment by marissaeinhorn** [comment]
Need to talk to the contractors about active directory permissions for vsphere folders.  I think the simplest solution will be to manually create the folders and have the contractors assign permissions for the folder.  Then build out the exercise in the existing folders for each new class.

Structure is:

> 17S
>> fallingStar
>>>team01
>>>>SISI
>>>>Students

>>>team02
>>>>SISI
>>>>Students

**Comment by marissaeinhorn** [comment]
Need to set static ip addresses for officeDC and opsDC.

**Comment by marissaeinhorn** [comment]
Outstanding template issues
- [ ] Static ips for domain controllers and workstations.  IPs were removed when cloning the templates, something wasnt set properly

**Comment by Timo604** [comment]
static IPs stay when cloning templates as long as the template can connect to the network the interface is configured for. Templates under FallingStar_Dev are set to `sisi_team1_office` or `sisi_team1_ops` which is available on morpheus but not neo.

`ipconfig /all` confirms the static IP is set along with the DNS Server and default gateway. Windows is autoconfiguring another IPv4 address because it seems to think the static IP is already in use. I don't think that'll be an issue once the VM is up in our deployment environment. Documenting in case it is an issue

**Comment by Timo604** [comment]
Disabling Windows Defender is done as a GPO instead of in each workstation template. Just verified in the gpreport.xml provided for the GPOs given to us that Defender AV is disabled and Defender FW has necessary ports open

**Comment by marissaeinhorn** [comment]
Student kali boxes network configurations set to `automatic dhcp` need to be changed to `manual` for networking to work as intended

**Comment by Timo604** [comment]
Building a Kali VM from the template on its own still has the network config set to manual

**Comment by marissaeinhorn** [comment]
- [ ] On OfficeDC the powershell script `Office_GPO_Users.ps1` is not executed as part of the templating process.  Also the file OfficeUsers.csv was not copied to `C:\Windows\Temp`
- [ ] OfficeDC ipaddress set improperly.  Currently is 10.0.20.42 needs to be 10.0.30.42
- [ ] OpsDC is an older version - ip address was 10.0.20.5, needs to be 10.0.20.42.  Scripts ran but were older versions 
- [ ] Ops Workstations have the wrong DNS server - needs to be 10.0.20.42 
- [ ] Office Workstations have the wrong DNS server - needs to be 10.0.30.42

**Comment by rylagek** [comment]
> * [ ]  On OfficeDC the powershell script `Office_GPO_Users.ps1` is not executed as part of the templating process.  Also the file OfficeUsers.csv was not copied to `C:\Windows\Temp`
> * [x]  OfficeDC ipaddress set improperly.  Currently is 10.0.20.42 needs to be 10.0.30.42
> * [x]  OpsDC is an older version - ip address was 10.0.20.5, needs to be 10.0.20.42.  Scripts ran but were older versions
> * [x]  Ops Workstations have the wrong DNS server - needs to be 10.0.20.42
> * [x]  Office Workstations have the wrong DNS server - needs to be 10.0.30.42

`main.pkr.hcl` updated for ips and dns servers

**Comment by rylagek** [comment]
Updated opsDC and put in FallingStarDev vSphere folder - not promoted but all files present

**Comment by marissaeinhorn** [comment]
- [ ] OfficeCatya scheduled task only runs when Administrator is logged in, should be run whether user is logged in or not.  Need to check whether generated meterpreter shell runs as clorn or Administrator.

**Comment by JohnLieman** [comment]
Powershell commands for configuring the Domain Controllers

**office.sisi DC (parent)** 
`
Import-Module ADDSDeployment`
`Install-ADDSForest ` -CreateDnsDelegation:$false ` -DatabasePath "C:\Windows\NTDS" ` -DomainMode "WinThreshold" ` -DomainName "office.sisi" ` -DomainNetbiosName "OFFICE" ` -SafeModeAdministratorPassword (convertto-Securestring -AsPlainText -Force asd@123) ` -ForestMode "WinThreshold" ` -InstallDns:$true ` -LogPath "C:\Windows\NTDS" ` -NoRebootOnCompletion:$true ` -SysvolPath "C:\Windows\SYSVOL" ` -Force ` -SkipPreChecks` 
`

**ops.office.sisi DC (child)**
`$User = "office.sisi\Administrator"`
`$PWord = ConvertTo-SecureString -String "P@ulconfigur3d@night" -AsPlainText -Force`
`$Credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $User, $PWord`
`Import-Module ADDSDeployment`
`
Install-ADDSDomain ` -NoGlobalCatalog:$false ` -Credential $Credential ` -DatabasePath "C:\Windows\NTDS" ` -DomainMode "ChildDomain" ` -InstallDns:$true ` -NewDomainName "ops" ` -NewDomainNetbiosName "OPS" ` -ParentDomainName "office.sisi" ` -SafeModeAdministratorPassword (convertto-Securestring -AsPlainText -Force asd@123) ` -LogPath "C:\Windows\NTDS" ` -NoRebootOnCompletion:$true ` -SiteName "Default-First-Site-Name" ` -SysvolPath "C:\Windows\SYSVOL" ` -Force:$true ` -SkipPreChecks
`

**Comment by JohnLieman** [comment]
`gpupdate /force` added to `Office_GPO_Users.ps1` and `Ops_GPO_Users.ps1`
OpsDC packer provisioner doesn't call `Ops_GPO_Users.ps1` at this time, not sure if that's intentional so I left it as is. 

DC passwords changed in `createFallingStar.yml`, `joinFallingStar.yml`, DC packer build file, and `child-dc.ps1`

**Comment by Timo604** [comment]
> * [ ]  OfficeCatya scheduled task only runs when Administrator is logged in, should be run whether user is logged in or not.  Need to check whether generated meterpreter shell runs as clorn or Administrator.

To add the user context during the packer provisioning isn't going to be possible. The user clorn doesn't yet exist since the template hasn't connected to the DC yet. If the user doesn't exist yet the task creation fails. 

One option is to have packer upload a script that will create the task. Ansible would run the script after the workstation has joined the domain so that windows understands the user context. 

Another option could be to have the task run as a service account NT AUTHORITY\SYSTEM. This should run without clorn needing to be logged in. I'm not sure how that might change the exercise, but I don't think it would since this context should be migrated to once the shell is caught.

**Comment by marissaeinhorn** [comment]
@Timo604 If the process runs as system and clorn is logged in later it wont impact the students ability to run the exercise.  Theyll be able to migrate to one of her processes once they've gained access and proceed without any problems

**Comment by marissaeinhorn** [comment]
Domain conflict on labnet fixed by assigning each team a vlan for the vsphere networks.

**Comment by JohnLieman** [comment]
Both DCs build successfully now.  Sysprep removed from both. 

**OfficeDC**
The `rename-computer` command was causing connection issues with winrm, preventing subsequent commands from running. Moved the command into the one-liner in the last provisioner to execute after the delayed shutdown. This issue doesn't occur on the OpsDC, and based on internet discussion seems to occur mostly in a domain environment. Since OpsDC isn't configured as part of the domain at this stage that may explain why it doesn't encounter this issue.

Packer attempted to execute its shutdown command and failed out while the delayed shutdown was still waiting, so I changed `shutdown /s /t 120` to `/t 45`

In the answer file Firstlogon synchronous commands, the `enable-winrm.ps1` script wasn't called after restart. Moved the script call to  `HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\RunOnce` key in registry.
